src/app/store/reducers/app-config.reducer.ts

```diff
@@ -24,6 +35,16 @@ export function reducer(
         loaded: true
       };
     }
+
+    case fromActions.GET_USER_DETAILS_SUCCESS: {
+      const userDetails = new UserModel(action.payload);
+      return {
+        ...state,
+        userDetails,
+        loaded: true,
+        loading: false,
```

Comment:
We seem to have only 1 loaded and loading flag on the entire state. What if multiple requests are fired at the same time (Not necessarily all to get user details, can be different requests)? The moment the first one finishes that will set these flags to true and false respectively, even though other requests may still be in flight. Have we thought about these potential issues?

Reply:
it is not the entire state its only app reducer... other requests are handled in different reducers... also due to poor implementation of this app we are not using  the loaded and loading to display spinner - so at this stage no potential issues with that